libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -78,28 +80,28 @@ AsmData::Statement InlineAssemblyParser::parseStatement()
 	// Simple instruction (might turn into functional),
 	// literal,
 	// identifier (might turn into label or functional assignment)
-	AsmData::Statement statement(parseElementaryOperation());
+	Statement statement(parseElementaryOperation());
 	switch (m_scanner->currentToken())
 	{
 	case Token::LParen:
 		return parseFunctionalInstruction(statement);
 	case Token::Colon:
 	{
-		if (statement.type() != typeid(AsmData::Identifier))
+		if (statement.type() != typeid(assembly::Identifier))
 			fatalParserError("Label name / variable name must precede \":\".");
-		string const& name = boost::get<AsmData::Identifier>(statement).name;
+		string const& name = boost::get<assembly::Identifier>(statement).name;
 		m_scanner->next();
 		if (m_scanner->currentToken() == Token::Assign)
 		{
 			// functional assignment
 			m_scanner->next();
-			unique_ptr<AsmData::Statement> value;
-			value.reset(new AsmData::Statement(parseExpression()));
-			return AsmData::FunctionalAssignment{{move(name)}, move(value)};
+			unique_ptr<Statement> value;
+			value.reset(new Statement(parseExpression()));
+			return FunctionalAssignment{{std::move(name)}, std::move(value)};
```

Comment:
Yes, some compiler complained about ambiguous identifiers. I think there is also some `boost::move`.